File: NodeJS/data/download_files.js
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch();
  const page = await browser.newPage();
  await page.goto('URL');

  const [ download ] = await Promise.all([
    page.waitForEvent('download'),
    page.click("Selector")  // Add your download link selector
  ]);
  await download.saveAs('downloaded_file.pdf');
  await browser.close();
})();

File: Python/save data/download_files.py
from playwright.sync_api import sync_playwright

with sync_playwright() as p:
    browser = p.chromium.launch()
    page = browser.new_page()
    page.goto("URL")
    with page.expect_download() as download_info:
        page.click("Selector")  # Add your download link selector
    download = download_info.value
    download.save_as("downloaded_file.pdf")
    browser.close()